lib/full-site-editing/templates.php

```diff
@@ -90,7 +90,7 @@ function gutenberg_register_wp_theme_taxonomy() {
 
 	register_taxonomy(
 		'wp_theme',
-		array( 'wp_template', 'wp_template_part' ),
+		array( 'wp_template', 'wp_template_part', 'wp_global_styles' ),
```

Comment:
There may be themes that have theme.json but aren't block-based themes and vice-versa, so we need to make sure that this relationship works as expected in those scenarios (a few lines above this there's a `gutenberg_supports_block_templates` check that makes this only work on block-based themes).

Reply:
Good catch I updated the condition used to register the taxonomy.

Reply:
Is it ok to register all of this or should we only register the individual pieces per each feature?

- register `wp_global_styles` if a theme has `theme.json` support
- register `wp_template` and `wp_template_part` if a theme has block-based templates

cc @vindl

Reply:
Here we are not registering wp_global_styles or wp_template we are registering wp_theme taxonomy and saying the taxonomy can be used on wp_global_styles, wp_template and wp_template_part block types. I think there is no issue in registering a taxonomy and saying it can be used in a CPT that is not yet available.